src/xcode/ENA/ENA/Source/Workers/SQLiteKeyValueStore.swift

```diff
@@ -23,26 +23,56 @@ class SQLiteKeyValueStore {
 	private let db: FMDatabase
 
 	/// - parameter url: URL on disk where the FMDB should be initialized
+	/// If any part of the init fails no Datbase will be created
 	init(with url: URL) {
-		let sqlStmt = """
-		CREATE TABLE IF NOT EXISTS kv (
-		    key TEXT UNIQUE,
-		    value BLOB
-		);
-		"""
-
 		db = FMDatabase(url: url)
-		db.open()
-		db.executeStatements(sqlStmt)
+		if !db.open() {
+			logError(message: "Database could not be opened")
+			return
+		}
+		initDatabase()
 	}
 
 	deinit {
 		db.close()
 	}
+	private func initDatabase(){
+		var key: String
+		if let keyData = loadFromKeychain(key: "secureStoreDatabaseKey") {
+			key = String(decoding: keyData, as: UTF8.self)
+		} else {
+			guard let generatedKey = generateDatabaseKey() else {
+				db.close()
+				return
+			}
+			key = generatedKey
+			if savetoKeychain(key: "secureStoreDatabaseKey", data: Data(key.utf8)) == noErr {
```

Comment:
Why not `key.data(using: .utf8)`? I have never used the `.utf8` property to get the data to be honest... Did I miss something? :D